src/Int.mo

```diff
@@ -58,4 +59,8 @@ module {
 
   public func pow(x : Int, y : Int) : Int { x ** y };
 
+  public class range(fromInclusive : Int, toExclusive : Int) {
```

Comment:
Thinking ahead about to finite types, exclusive bounds won't let us specify a range up to the last element of a finite type).
Should we consider an API like

public class range(fromInclusive : T, {#Exc : T; #Inc : T) {..}

I guess the alternative is to provide 

public class rangeInc(fromInclusive : T, toExclusive : T) {..}


Also should the range count down if the bound is below the range? I don't know what we do at the moment.

Reply:
That's a very good point. This would be my preference to simplify the most common case:

```motoko
public class range(fromInclusive : T, toExclusive : T) : T { ... }

public class rangeInclusive(from : T, to : T) { ... }
```

If it's potentially confusing for a finite type, we could rename `range()` to `rangeExclusive()` for that module to make it an explicit choice. 

We could also simplify the most likely use case by including `allValues()` functions for `Nat8`, `Int8`, `Bool`, `Compare`, etc.